### PR TITLE
Pc/fbprophet

### PR DESCRIPTION
**Pull request by Blaizzy**
# Description

Please include a summary of the changes and the related issue.

__Related to:__ `(ClickUp/JIRA task link)`

---

## ❔ This change

- [x] adds a new feature
- [ ] fixes breaking code
- [ ] is cosmetic (refactoring/reformatting)

---

## ✔️ Pre-merge checklist

Mark which items have been completed:

- [x] Refactored code ([sourcery](https://sourcery.ai/))
- [x] Tested code locally
- [x] Added code to GitHub tests ([notebooks](workflows/test-notebooks.yml), [scripts](workflows/test-scripts.yml))
- [x] Updated GitHub [README](../README.md)

### 🪝 Run the following to both [pre-commit](https://pre-commit.com/) hooks and JupyterLab (turn on `format on save`)

- [x] Stripped outputs from notebooks - ([nbstripout](https://pypi.org/project/nbstripout/))
- [x] Sorted imports - ([isort](https://pycqa.github.io/isort/))
- [x] Formatted code - ([back](https://github.com/psf/black))

### 📄 Updated documentation  

Share the link of the documentation branch with changes: ([link](https://app.gitbook.com/o/-MGlxksZUK9_I6_8WMJ8/s/-MT0sYKbymfLAAtTq4-t/~/changes/xP3gw0F3bebxKUlFil5G/integrations-and-supported-tools/model-training/prophet))

- [x] Updated content in the [docs](https://docs.neptune.ai) (Sabine/@normandy7)
- [x] Documentation all relevant docs index pages (e.g., [Examples](https://docs.neptune.ai/getting-started/examples), [Integrations intro](https://docs.neptune.ai/integrations-and-supported-tools/intro), [Model training index](https://docs.neptune.ai/integrations-and-supported-tools/model-training)) (Sabine/@normandy7)

---

## 🧪 Test Configuration

- OS: macOS Monterey v12.4
- Python version: 3.8.13
- Neptune version: 0.16.2
- Affected libraries with version: 

---

## ✔️ Post-merge checklist

These tasks should be done after the code is merged to main. Please create a subtask under the main ClickUp task to have these on your radar and share the subtask here.

__Post-merge subtask:__ `(ClickUp/JIRA task link)`

- [ ] Update relevant blogs with impacted code/functionality (ClickUp task name: Update prophet articles assigned to @SiddhantSadangi )
- [ ] Add functionality under the Neptune column of the `competitor comparison sheet` (Coordinate with @SiddhantSadangi)
- [ ] Work with marketing team to share updates on Social Media (Contact @Patrycja-J/@jakubczakon)

### CODE REVIEW

**Comment by SiddhantSadangi** [comment]
@Blaizzy lets hold merging this until `neptune-prophet==1.0.0` is released. 
Meanwhile, can you check the failing tests if anything looks out of the ordinary, and add more details to the PR checklist?

**Comment by Blaizzy** [comment]
> @Blaizzy lets hold merging this until `neptune-prophet==1.0.0` is released. Meanwhile, can you check the failing tests if anything looks out of the ordinary, and add more details to the PR checklist?

It was a mistake during release they put 1.0.0 while it's supposed to be 0.1.0.

We can proceed.

**Comment by Blaizzy** [comment]
@SiddhantSadangi tests are passing but some other tests are failing for something not very clear to me.

>   src/checkdep_freetype2.c(5): fatal error C1189: #error:  "FreeType version 2.3 or higher is required. You may set the MPLLOCALFREETYPE environment variable to 1 to let Matplotlib download it."
...
> RuntimeError: main thread is not in main loop
Tcl_AsyncDelete: async handler deleted by the wrong thread

**Comment by Blaizzy** [comment]
The PR is ready for review.

@normandy7 and @SiddhantSadangi  I pass it over to you guys.

Let me know if there are any further changes needed 👍

File: integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e37c2302",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "\n",
+    "This quickstart will show you how to (using just single function) log FBprophet:\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb:18]: It should be referred to as just "Prophet"

File: integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8f521acb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run = neptune.init(\n",
+    "    project='common/fbprophet-integration', \n",
+    "    api_token=\"ANONYMOUS\",\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb:88]: use `api_token=ANONYMOUS_API_TOKEN`

File: integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8f521acb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run = neptune.init(\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb:86]: use `neptune.init_run()`

File: integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+    "* **where you want to send your data**: your Neptune `project`.\n",
+    "\n",
+    "At this point you have new run in Neptune. For now on you will use `run` to log metadata to it.\n",
+    "\n",
+    "---\n",
+    "\n",
+    "**Note**\n",
+    "\n",
+    "\n",
+    "Instead of logging data to the public project `'common/sklearn-integration'` as an anonymous user 'neptuner' you can log it to your own project.\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb:112]: sklearn artifact

File: integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+    "\n",
+    "To do that:\n",
+    "\n",
+    "1. Get your [Neptune API token](https://docs.neptune.ai/getting-started/installation#get-api-token)\n",
+    "2. Pass the token to ``api_token`` argument of ``neptune.init()`` method: ``api_token=YOUR_API_TOKEN``\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb:117]: use single backticks for inline code: \`api_token\`

File: integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+    "3. Pass your project to the ``project`` argument of ``neptune.init()``.\n",
+    "\n",
+    "For example:\n",
+    "\n",
+    "```python\n",
+    "neptune.init(project=\"YOUR_WORKSPACE/YOUR_PROJECT\",\n",
+    "             api_token=\"YOUR_API_TOKEN\")\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb:124]: We will remove passing user's own API token from examples at some point, but for now we can keep it

File: integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "98c5a2cb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import neptune_prophet.impl as npt_utils\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb:195]: How do we want to do this? It's inconsistent with the examples in the docstrings.

cc: @twolodzko

File: integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import neptune.new as neptune
+import neptune_prophet.impl as npt_utils
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:2]: Is this how we want to do this? (Same question as for the ipynb example)

**Review by normandy7 — COMMENTED** [review]
Left some requests; would be great if they can be taken into account in the new sample code.

File: integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py (hunk 8 above)
Comment (twolodzko) [reply integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:2]: No, it should be now importable as all the other integrations.

File: integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    project="common/fbprophet-integration",
+    api_token="ANONYMOUS",
+    tags=["fbprophet", "additional regressors", "script"],  # optional
+)
+
+df = pd.read_csv(
+    "https://raw.githubusercontent.com/facebook/prophet/master/examples/example_wp_log_R.csv"
+)
+
+df["cap"] = 8.5
```
Comment (twolodzko) [inline integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:16]: What is this?

File: integrations-and-supported-tools/sklearn/notebooks/Neptune_Scikit_learn.ipynb [hunk 10]
```diff
@@ -1055,6 +1055,11 @@
    "provenance": [],
```
Comment (twolodzko) [inline integrations-and-supported-tools/sklearn/notebooks/Neptune_Scikit_learn.ipynb:1055]: Why this goes together? Shouldn't it go as a separate PR?

Comment (Blaizzy) [reply integrations-and-supported-tools/sklearn/notebooks/Neptune_Scikit_learn.ipynb:1055]: I don't know what happened exactly, but this should not be here indeed.

File: integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb (hunk 4 above)
Comment (Blaizzy) [reply integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb]: done

File: integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb (hunk 5 above)
Comment (Blaizzy) [reply integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb:117]: It's a string, it doesn't change anything. Users will only see text/

File: integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py (hunk 9 above)
Comment (Blaizzy) [reply integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:16]: That's market size, it's part of the example in FBprophet docs.

https://facebook.github.io/prophet/docs/saturating_forecasts.html#forecasting-growth

I don't have any strong opinions on keeping it, so if it makes it easier for the users, we can remove that specific line.

File: integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py (hunk 8 above)
Comment (Blaizzy) [reply integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:2]: I'm getting the following error
`ModuleNotFoundError: No module named 'neptune.new.integrations.prophet'`

When I  run 
`import neptune.new.integrations.prophet as npt_utils `

Comment (twolodzko) [reply integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:2]: OK, it seems that we didn't release the client version that has the integration added. Still, in the docs we probably want the proper way that would be available as soon as we release the current client.

File: integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py (hunk 9 above)
Comment (twolodzko) [reply integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:16]: OK, wasn't aware of that. Maybe we could add a code comment to say what it is? It looks strange that this line appears out of the blue in the code.

File: integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py (hunk 8 above)
Comment (Blaizzy) [reply integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:2]: It's probably not released yet. :

Comment (normandy7) [reply integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:2]: I'm still a bit confused. In the docs examples, which are based on the docstrings, there is no `import neptune.new.integrations.prophet as npt_utils`. Will it be needed or not? :)

Comment (twolodzko) [reply integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:2]: @normandy7 no, it will be importable like other integrations, so `from neptune.new.integrations.prophet import ...` , we just need to release the new client version that has the changes. In the docs we should use the "proper" way.

**Comment by normandy7** [comment]
I tried out the example notebook in Colab, but I get an error when running the following:

```
import neptune.new.integrations.prophet as npt_utils

run["prophet_summary"] = npt_utils.create_summary(model=model, df=df, fcst=forecast)
```

```
AttributeError: 'Line2D' object has no property 'auto_ylims'
```

Could you verify that it works on your end?

**Comment by Blaizzy** [comment]
@normandy7, that is weird.

It works normally on my local machine, but in colab, it indeed throws an error

**Comment by Blaizzy** [comment]
@normandy7  found the issue.

Colab, by default, is not on the latest `statsmodels v0.13.2` library, which causes the error.
It can be fixed by updating it:

`!pip install -U statsmodels`

**Comment by twolodzko** [comment]
@Blaizzy @normandy7 Prophet dependencies were updated and it was released. It should work for now, but for Conda, we might need to wait longer till it catches up.

**Comment by Blaizzy** [comment]
Great to hear, thanks @twolodzko !

**Comment by Blaizzy** [comment]
Back to you @SiddhantSadangi 👍🏾

**Review by SiddhantSadangi — APPROVED** [review]
lgtm

File: integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb [hunk 11]
```diff
@@ -0,0 +1,433 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8f521acb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run = neptune.init_run(\n",
+    "    project=\"common/fbprophet-integration\",\n",
+    "    api_token=\"ANONYMOUS\",\n",
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/notebooks/Neptune_prophet.ipynb:90]:
```suggestion
    "    api_token=neptune.ANONYMOUS_API_TOKEN,\n",
```

File: integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
+import neptune.new as neptune
+import neptune.new.integrations.prophet as npt_utils
+import pandas as pd
+from prophet import Prophet
+
+run = neptune.init_run(
+    project="common/fbprophet-integration",
+    api_token="ANONYMOUS",
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/scripts/Neptune_prophet.py:8]:
```suggestion
    api_token=neptune.ANONYMOUS_API_TOKEN,
```

File: integrations-and-supported-tools/prophet/scripts/Neptune_prophet_more_options.py [hunk 13]
```diff
@@ -0,0 +1,42 @@
+import neptune.new as neptune
+import neptune.new.integrations.prophet as npt_utils
+import pandas as pd
+from prophet import Prophet
+
+run = neptune.init_run(
+    project="common/fbprophet-integration",
+    api_token="ANONYMOUS",
```
Comment (normandy7) [inline integrations-and-supported-tools/prophet/scripts/Neptune_prophet_more_options.py:8]:
```suggestion
    api_token=neptune.ANONYMOUS_API_TOKEN,
```

**Review by normandy7 — APPROVED** [review]
